packages/core/src/BlockNoteEditor.ts

```diff
@@ -266,7 +250,7 @@ export class BlockNoteEditor<
    * @param reverse Whether the blocks should be traversed in reverse order.
    */
   public forEachBlock(
-    callback: (block: Block) => boolean,
+    callback: (block: BaseBlock) => boolean,
```

Comment:
So the `BaseBlock` thing might be a little confusing, but you can understand like this.
- A specific Block, selected by something: `Block`
- A generic Block, such as children or callback: `BaseBlock`

A `BaseBlock` is a `Block` that has no opinion on typing. The `Type` and `Props` of a `BaseBlock` is `string` and `Record<string, string>` since we don't know about `BaseBlock`. These properties apply to children or callback blocks.